### PR TITLE
CSE-851 multi field mapping

### PR DESCRIPTION
**Pull request by voodoods**
**Jira Issue**

https://8select.atlassian.net/browse/CSE-851

**Summary**

- enabled selection of multiple shopware standard-fields in attribute mapping
- refactored processing of configurator group values and property filter values for export
- prevent multiple standard `s_articles**` fields from being selected in attribute mapping since export can only process multiple values for configurator groups and property filters
- if both configurator group values and filter option values exist on a mapped field, the filter option values will be ignored to prevent multiple values in a variant property

**Checklist**

- [x] PR title is prefixed with Jira Issue Id - e.g. CSE-42
- [x] Add feature / bug label
- [x] Unit tests for new / changed logic exist OR no logic changes are passing

**Example Export CSV file:** 
https://productfeed-prod.staging.8select.io.s3.amazonaws.com/438d6cfe-d9f4-402b-b9cb-cb6070300ddf/product_feed/2018/07/27/438d6cfe-d9f4-402b-b9cb-cb6070300ddf_product_feed_1532698764789.csv

**Screenshots:** 

bildschirmfoto 2018-05-30 um 13 21 57

bildschirmfoto 2018-05-30 um 13 22 18

**Disable standard fields for multiselection:**

42376117-4198e70a-811e-11e8-8fe9-eaa151eaa184

### CODE REVIEW

**Comment by voodoods** [comment]
@Wytrykus I found out that you actually could deselect the currently set value when you click on it again. This also applies for every other item. But the user wouldn't grasp that, so I don't think thats a good solution from a UX point of view.

New solution: I'll now show an alert that explains to the user that he can only multiselect actual config or filter property fields:

bildschirmfoto 2018-07-06 um 13 02 25

At the same time the value of the field gets cleared so that the user can make a new selection. This won't break the input behavior for the user. What do you think?

**Comment by Wytrykus** [comment]
@voodoods  Please rebase again

**Comment by voodoods** [comment]
@Wytrykus done

File: CseEightselectBasic/Components/FieldHelper.php [hunk 1]
```diff
@@ -95,22 +95,24 @@ private static function getValue($article, $field)
...
-        $filterGroup = self::getGroupOrFilterAttribute('filter', $field);
-
-        if ($filterGroup) {
-            $filterId = explode('id=', $filterGroup)[1];
-
-            return self::getFilterValues($article['articleID'], $filterId);
+        $attrGroup = self::getGroupOrFilterAttribute($field);
+
+        if ($attrGroup) {
+            $valuesTreshold = [];
```
Comment (Wytrykus) [inline CseEightselectBasic/Components/FieldHelper.php]: Why is this variable named `valuesTreshold`? Doesn't make sense to me.

Comment (voodoods) [reply CseEightselectBasic/Components/FieldHelper.php]: Good point, I misinterpreted the translation for the word `treshold`. Kept it simple now and just named it `$values`.

File: CseEightselectBasic/CseEightselectBasic.php [hunk 2]
```diff
@@ -619,6 +619,13 @@ private function initAttributes()
                 'eightselectAttributeLabel'      => 'EAN-CODE',
                 'eightselectAttributeLabelDescr' => 'Standardisierte eindeutige Materialnummer nach EAN (European Article Number) oder UPC (Unified Product Code).',
                 'shopwareAttribute'              => 's_articles_details.ean',
+
```
Comment (jbreckel) [inline CseEightselectBasic/CseEightselectBasic.php]: please remove additional new line

File: CseEightselectBasic/CseEightselectBasic.php [hunk 3]
```diff
@@ -619,6 +619,13 @@ private function initAttributes()
...
+            ],
+            [
+                'eightselectAttribute'           => 'model',
+                'eightselectAttributeLabel'      => 'PRODUKT-MODELL ID',
+                'eightselectAttributeLabelDescr' => 'Ein Modell bezeichnet das "Grundprodukt". ID kann Name oder Nummer sein. Ein Modell kann in mehreren Ausführungen vorkommen, die sich z.B. in Größe, Farbe oder Muster unterschieden. Beispiel: Das Modell "Arie" (1234) gibt es in den 3 Farben: "blau", "rot", "gelb" (567) und in den 4 Größen: "S", "M", "L", "XL" (89), also 12 Ausführungen. Jede Ausführung hat eine eindeutige Artikelnummer.',
+                'shopwareAttribute'             => '-',
```
Comment (jbreckel) [inline CseEightselectBasic/CseEightselectBasic.php]: please add a space to even the fat arrows, for all cases in this file

**Comment by voodoods** [comment]
@akleiber 

I reworked the logic for when both **configurator group options**  and **filter options** are mapped to an eightselect attribute field: 
- the script will now first check, if values for the configurator group options are available; if so, they will be written into the field while ignoring existing values for filter options
- only if there are no values for configurator group options available on the given product, the filter option values will be written into the field instead

This works as long as there are not multiple configurator group options mapped to a single field that all have values - which shouldn't be practical anyway. Please check 👍

**Comment by voodoods** [comment]
@akleiber fixed it, you can now start testing it manually

**Pull request merged**